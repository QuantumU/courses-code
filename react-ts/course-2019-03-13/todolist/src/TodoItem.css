.todo {
  margin: 5px;
  padding: 5px;
  background-color: lightblue;
}

.todo.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}