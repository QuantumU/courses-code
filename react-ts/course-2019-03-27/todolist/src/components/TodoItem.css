.todo-item {
  background-color: lightgreen;
  padding: 8px;
  margin: 8px;
}

.todo-item.completed {
  background-color: lightgrey;
  text-decoration: line-through;
}
