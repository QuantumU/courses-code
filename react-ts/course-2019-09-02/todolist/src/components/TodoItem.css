.todo {
  background-color: lightcoral;
  margin: 5px;
  padding: 5px;
}

.todo.completed {
  background-color: lightgrey;
}

.todo.completed>.label {
  text-decoration: line-through;
}
