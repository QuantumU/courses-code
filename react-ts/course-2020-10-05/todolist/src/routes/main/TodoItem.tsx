import React, { memo } from "react";
import styled from "styled-components";

type TodoItemProps = {
  completed: boolean;
  title: string;
  onToggle: () => void;
  onDelete: () => void;
};

const TodoItemLi = styled.li<{ completed: boolean }>`
  margin: 5px;
  padding: 5px;
  background-color: ${(props) =>
    props.completed ? "lightgrey" : "lightcoral"};
  text-decoration: ${(props) => (props.completed ? "line-through" : "none")};
`;

const DeleteBtn = styled.button`
  background-color: lightblue;
`;

function TodoItem(props: TodoItemProps) {
  return (
    <TodoItemLi
      className={"todoitem" + (props.completed ? " completed" : "")}
      onClick={() => {
        props.onToggle();
      }}
      completed={props.completed}
    >
      {props.completed ? "DONE" : "TODO"}: {props.title}
      <DeleteBtn
        onClick={(event) => {
          event.stopPropagation();
          props.onDelete();
        }}
      >
        X
      </DeleteBtn>
    </TodoItemLi>
  );
}

export default memo(TodoItem);
