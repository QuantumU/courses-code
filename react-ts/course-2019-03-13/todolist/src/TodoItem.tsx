import React, { memo, Component } from 'react';
import TodoType from './TodoType';

import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onCompleted: () => void;
};

class TodoItem extends Component<TodoItemProps, {}> {

  shouldComponentUpdate(newProps: TodoItemProps) {
    return this.props.todo !== newProps.todo;
  }

  render = () => {
    return (
      <div
        className={`todo${this.props.todo.completed ? ' completed' : ''}`}
        onClick={() => {
          this.props.onCompleted();
        }}
      >
        {this.props.todo.completed ? 'DONE: ' : 'TODO: '}{this.props.todo.title}
      </div>
    );
  };
}

export default TodoItem;
