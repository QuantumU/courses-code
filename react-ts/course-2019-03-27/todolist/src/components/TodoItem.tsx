import React, { PureComponent } from 'react';
import TodoType from '../TodoType';
import './TodoItem.css';

type TodoItemProps = {
  todo: TodoType;
  onToggle: (id: number) => void;
};

type TodoItemState = {};

class TodoItem extends PureComponent<TodoItemProps, TodoItemState> {
  render() {
    // "todo-item" or "todo-item completed"
    return (
      <div
        className={
          this.props.todo.completed ? 'todo-item completed' : 'todo-item'
        }
        id="abc"
        onClick={() => {
          this.props.onToggle(this.props.todo.id);
        }}
      >
        {this.props.todo.title}
      </div>
    );
  }
}

export default TodoItem;
