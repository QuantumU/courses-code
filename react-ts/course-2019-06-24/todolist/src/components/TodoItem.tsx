import React from "react";
import { TodoType } from "../TodoType";
import "./TodoItem.css";

type TodoItemProps = {
  todo: TodoType;
  toggleTodo: (id: number) => void;
  deleteTodo: (id: number) => void;
};

const TodoItem = (props: TodoItemProps) => {
  return (
    <li
      key={props.todo.id}
      className={props.todo.completed ? "todo completed" : "todo"}
      onClick={() => {
        props.toggleTodo(props.todo.id);
      }}
    >
      {props.todo.completed ? "DONE: " : "TODO: "}
      {props.todo.title}
      <button onClick={() => {
        props.deleteTodo(props.todo.id)
      }}>X</button>
    </li>
  );
};

export default TodoItem;
