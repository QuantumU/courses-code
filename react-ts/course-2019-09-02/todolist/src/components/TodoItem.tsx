import React, { Component } from "react";
import TodoType from "../TodoType";
import "./TodoItem.css";
import Button from "@material-ui/core/Button";

type TodoItemProps = {
  todo: TodoType;
  onCompleted: (id: number) => void;
  onDelete: (id: number) => void;
};

class TodoItem extends Component<TodoItemProps> {
  render() {
    return (
      <div
        className={this.props.todo.completed ? "todo completed" : "todo"}
        onClick={this.handleClick}
      >
        <span className="label">
          {this.props.todo.completed ? "DONE" : "TODO"}: {this.props.todo.title}
        </span>{" "}
        <Button onClick={this.handleDelete} variant="contained" color="primary">
          X
        </Button>
      </div>
    );
  }

  handleClick = () => {
    this.props.onCompleted(this.props.todo.id);
  };

  handleDelete = (clickEvent: React.MouseEvent<HTMLButtonElement>) => {
    clickEvent.stopPropagation();
    this.props.onDelete(this.props.todo.id);
  };
}

export default TodoItem;
