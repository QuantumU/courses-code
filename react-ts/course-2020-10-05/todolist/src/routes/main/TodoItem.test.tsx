import React from "react";
import TodoItem from "./TodoItem";
import { render } from "@testing-library/react";

it("matches the snapshot", () => {
  const instance = render(
    <TodoItem
      title="foo"
      completed={false}
      onToggle={() => {}}
      onDelete={() => {}}
    />
  );
  expect(instance.container).toMatchSnapshot();
});
