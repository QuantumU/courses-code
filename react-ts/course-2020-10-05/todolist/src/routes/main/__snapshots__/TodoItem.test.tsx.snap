// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <li
    class="sc-bdnylx jhzCVK todoitem"
  >
    TODO
    : 
    foo
    <button
      class="sc-gtssRu fdcDti"
    >
      X
    </button>
  </li>
</div>
`;
